homeassistant/core.py

```diff
@@ -262,8 +262,8 @@ def async_block_till_done(self):
             self._pending_tasks.clear()
             if pending:
                 yield from asyncio.wait(pending, loop=self.loop)
-            else:
```

Comment:
The reason we yield inside the while loop is so that we give scheduled callbacks (which are called with `loop.call_soon`) time to execute as they are not added to `self._pending_tasks`.